src/collection.cpp

```diff
@@ -6533,6 +6524,23 @@ Option<bool> Collection::parse_facet(const std::string& facet_field, std::vector
                     break;
                 }
             } else if(range_string[index] == ',' && range_open == false) {
+                if(index > 0 && range_string[index - 1] != ']') {
+                    //top_k string value
+                    auto top_k_str = range_string.substr(startpos, index - startpos);
+                    if(top_k_str.find("top_k") != std::string::npos) {
+                        std::vector<std::string> results;
+                        StringUtils::split(top_k_str, results, ":");
+
+                        if(results.size() != 2 || ((results[1] != "true") && (results[1] != "false"))) {
```

Comment:
Redundant parenthesis. around `(results[1] != "true")` and `(results[1] != "false")`